gpflow/conditionals/mo_conditionals.py

```diff
@@ -96,12 +95,12 @@ def single_gp_conditional(t):
     rmu, rvar = tf.map_fn(single_gp_conditional, (Kmms, Kmns, Knns, fs, q_sqrts),
                           (default_float(), default_float()))  # [P, N, 1], [P, 1, N, N] or [P, N, 1]
 
-    fmu = rollaxis_left(rmu[..., 0], 1)  # [N, P]
+    fmu = rollaxis_left(tf.squeeze(rmu, axis=-1), 1)  # [N, P]
```

Comment:
tf.squeeze() includes a check that this axis has length 1, it's good to have even a small additional safe guard.

Reply:
i'd suggest leaving this kind of change to a separate PR, it makes reviewing a lot easier as I need to look out for fewer things

Reply:
well it kind of goes with the asserting shapes - it's asserting that that dimension has length 1!